### ISSUE TITLE
Support Node.js Express?

### ISSUE THREAD

**Issue by aseemk** [sstephenson/eco#12]
Hey there,

Great work on this stuff! Just one request: is it possible to use this as a template engine in Express?

I'm not sure exactly what that entails, but I don't think it's much; I'm pretty sure it's just a matter of exposing the expected APIs. I can ask the Express author(s) for details on this if you would like.

Here's the sample app I made to test this: https://gist.github.com/927144 (Check out the comments to see the error message.)

Thanks in advance!

Cheers,
Aseem

**Comment by mapmarkus** [comment]
I use **eco** in **express** (v 2.3.3) registering `.html` like this

``` coffeescript
app.set 'view engine', 'html'
app.register '.html', 
   compile: (str, options) ->
      return (locals) ->
         eco.render str, locals
```

**Comment by aseemk** [comment]
Thanks @yizzreel -- that works like a charm! I'll keep this issue open; it would still be great for Eco to support Express out-of-the-box.

**Comment by mapmarkus** [comment]
If you take a quick look at https://github.com/sstephenson/eco/blob/master/src/eco/compiler.coffee#L6, you'll see that the problem registering _eco_ as a processor in _expresso_ is the call to the method `compile`, which in _eco_ returns a string to be interpreted by the class `Function()` (that takes the list of arguments and the function body string in the constructor), while _express_ expects it to return a callback, so it tries to use that returned string as a the expected callback, hence the error you're getting in your gist page.

I think _express_ recently changed the default method for view rendering from `render` to `compile`, so this error might be new, since _express_ and _eco_ are being used together in https://github.com/sstephenson/stitch (not sure which versions though, I'm still kinda new in all this coffeescript awesomeness).

**Comment by aseemk** [comment]
Thanks for looking into it! I'm not knowledgeable enough to understand then what the right course of action is w.r.t. backwards compatibility, etc., but one quick comment:

> since express and eco are being used together in https://github.com/sstephenson/stitch

Hmm, are you sure? I see the sample code mentions Stitch is to be used with Express, but I don't see any reference to Eco anywhere. The source also isn't an Express app.

**Comment by mapmarkus** [comment]
See https://github.com/sstephenson/stitch/blob/master/src/stitch.coffee#L20. And, it's true, express is not part of stitch, but they work together nicely.

**Comment by aseemk** [comment]
Apologies if I'm still being dense, but it still doesn't seem to use Express's view system for compiling Eco -- which is what my issue is about. Instead, Stitch seems to compile Eco directly.

Cool addition in your fork. =) Plan on submitting it as a pull request? Btw, maybe it would be better named `express` to make it clear this is the property to use with Express?

**Comment by mapmarkus** [comment]
The pull is already submitted :) https://github.com/sstephenson/eco/pull/19 (it's also in this comment thread, above your last comment). I used the less intrusive name I could think of, to make the smallest change possible.

**Comment by sstephenson** [comment]
Fixed in 5d62cdd9db455ea6ac3e4a0634779f8161ea5def.

``` coffeescript
express = require "express"
eco = require "eco"

app = express.createServer()
app.register ".eco", eco
```

**Comment by aseemk** [comment]
Awesome! Could you bump the version number and npm publish this? Thanks!
